openeo/testing/results.py

```diff
@@ -8,8 +8,11 @@
 from pathlib import Path
 from typing import List, Optional, Union
 
+import numpy as np
 import xarray
 import xarray.testing
+from scipy.spatial import ConvexHull
```

Comment:
This makes scipy a runtime dependency (now it's just a test dependency). Scipy is quite a heavy dependency (in terms of compiled extensions and additional transitive dependencies), which we we generally want to avoid.

If we keep it as optional dependency, this should be documented, e.g. here:
https://github.com/Open-EO/openeo-python-client/blob/c6bc1c91ff7ba542f044c5ffea6d4ac3379f242e/docs/installation.rst?plain=1#L82-L95

Or, is it actually necessary to have scipy as hard dependency here? Is that convex hull feature vital for the reporting quality? I think there could be a simpler fallback fraction-of-area calculation if ConvexHull is not available

Reply:
for example: you use the convex hull to estimate the impacted area of coordinates with differences, but a much simpler/coarser approximation is just taking the min/max values of these coordinates to get the bounding box. Which does not require scipy